✨ Add mongodb32 to withOpenshiftServices

grdryn opened the pull request:
This is needed for fh-db

grdryn committed:
✨ Add mongodb32 to withOpenshiftServices

This is needed for fh-db

pmccarthy commented on vars/withOpenshiftServices.groovy:
```diff
@@ -100,7 +100,7 @@ Map<String, String> getNames(List<String> services) {
     List<String> out = []
-    if (services.contains('mongodb')) {
+    if (services.contains('mongodb') || services.contains('mongodb32')) {
         out.add("MONGODB_HOST=${names[services.indexOf('mongodb')]}")
```
We're referencing mongodb here. Should this not cater for both mongodb and mongodb32?

grdryn replied:
@pmccarthy It should, yes. I'll change it.

For anyone else reading this, the reason it still worked in my testing, even though I forgot to change it is interesting:

The `.indexOf` method on a List [comes from Java](http://docs.oracle.com/javase/8/docs/api/java/util/List.html#indexOf-java.lang.Object-)
> Returns:
    the index of the first occurrence of the specified element in this list, _**or -1 if this list does not contain the element**_

In Groovy though, one can use negative indices with the [subscript operator](http://groovy-lang.org/operators.html#subscript-operator) to count from the end of the list.

I had a list of 1, so this unintentionally worked. When changing this, I'll see if there's another way to do this that will be less scary.

grdryn replied:
@pmccarthy I've fixed that now, would you mind taking another look?

grdryn committed:
🐛 Rewrite away bug in List indexing

I rewrote the env method in withOpenshiftServices to no longer have
the bug where it would use List#indexOf, which could lead to -1 if the
element didn't exist, then we used that to index into a different
List, which would give the last element.

grdryn committed:
🎨 'getNames' in withOpenshiftServices is just map

pmccarthy reviewed (approved):
Validated as part of https://jenkins-wendy.ci.feedhenry.org/job/feedhenry/job/fh-db/job/PR-34/50/console
